pkg/api/metrics.go

```diff
@@ -73,6 +74,10 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext, reqDto dtos.MetricReq
 	if !expr {
 		resp, err = tsdb.HandleRequest(c.Req.Context(), ds, request)
 		if err != nil {
+			if errors.Is(err, tsdberrs.BadRequest{}) {
```

Comment:
I cannot see any code referring to creating a `tsdberrs.BadRequest` so this should probably be removed.